File: backend/app/schemas/payment.py
from datetime import datetime

from app.models.order import PaymentMethod
from app.models.payment import PaymentStatus
from app.schemas.common import CamelModel


class PaymentOut(CamelModel):
    id: str
    order_id: str
    method: PaymentMethod
    status: PaymentStatus
    transaction_id: str
    created_at: datetime
    updated_at: datetime
    # Presente só logo após criar/reiniciar o pagamento via Mercado Pago —
    # URL do Checkout Pro pro front redirecionar o navegador. None quando
    # não há redirecionamento (dinheiro, ou pagamento já aprovado).
    init_point: str | None = None

File: src/types/payment.ts
import type { PaymentMethod } from "./tenant";

export type PaymentStatus = "pendente" | "aprovado" | "recusado";

export interface Payment {
  id: string;
  orderId: string;
  method: PaymentMethod;
  status: PaymentStatus;
  transactionId: string;
  createdAt: string;
  updatedAt: string;
  /** Só vem preenchido logo após criar/reiniciar o pagamento via Mercado Pago. */
  initPoint?: string | null;
}
